Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -18,38 +19,61 @@ public final class DiagnosticConsoleWriter: DiagnosticFormattingConsumer {
     var outputStream: TextOutputStream
     public var formattingOptions: DiagnosticFormattingOptions
     private var diagnosticFormatter: DiagnosticConsoleFormatter
+    private var problems: [Problem] = []
 
     /// Creates a new instance of this class with the provided output stream and filter level.
     /// - Parameter stream: The output stream to which this instance will write.
     /// - Parameter filterLevel: Determines what diagnostics should be printed. This filter level is inclusive, i.e. if a level of ``DiagnosticSeverity/information`` is specified, diagnostics with a severity up to and including `.information` will be printed.
     @available(*, deprecated, message: "Use init(_:formattingOptions:) instead")
     public convenience init(_ stream: TextOutputStream = LogHandle.standardError, filterLevel: DiagnosticSeverity = .warning) {
-        self.init(stream, formattingOptions: [])
+        self.init(stream, formattingOptions: [], baseURL: nil, highlight: nil)
     }
 
     /// Creates a new instance of this class with the provided output stream.
     /// - Parameter stream: The output stream to which this instance will write.
-    public init(_ stream: TextOutputStream = LogHandle.standardError, formattingOptions options: DiagnosticFormattingOptions = []) {
+    public init(
+        _ stream: TextOutputStream = LogHandle.standardError,
+        formattingOptions options: DiagnosticFormattingOptions = [],
+        baseURL: URL? = nil,
+        highlight: Bool? = nil
```

Comment:
That's a great point, thank you for highlighting this! I update the pull request with the changes mentioned to make the abstraction improvements easier later on.

Reply:
> I don't think it's necessary (or even preferable) to fix the abstraction in this PR but I would like to make it easier to fix the abstraction later. To make that easier I think it would be good to keep the previous public initializer unchanged and add a new internal initializer that accept all 4 arguments. That way, we can change how the base URL and highlighting are configured in the future without breaking public API.

@d-ronnqvist I noticed that a public initializer is required if we want to set the `baseUrl` in both `ConvertAction` and `IndexAction` since they are defined in `SwiftDocCUtilities`.
This is currently failing the build as it needs to be updated. I just wanted to confirm that this is what we want or we want to make it available to those actions? After I get confirmaiton I'll push a commit to fix this and the build should pass! 🙂